src/lib/isEmail.js

```diff
@@ -141,11 +141,15 @@ export default function isEmail(str, options) {
     }
   }
 
+  if (user.charAt(user.length - 1) === '-') {
```

Comment:
Is `-` the only invalid _terminal_ char? 🤔

Reply:
I tried to add these tests in invalid emails
```
'abc-@mail.com',
'abc..def@mail.com',
'..abc@mail.com',
'abc@mail..com',
'.abc@mail.com',
'abc.def@mail.c',
'abc.def@ma@il.c',
 'abc.def@mail',
 'abc.def@mail..com',
 'abc.def@mail#archive.com',
```
The one that failed is the first one `abc-@mail.com`. That is why I checked if an name section ends with `-`, I make it invalid. 
I was also looking at an appropriate place to add... 
If there is a place where it can be added in the regex, that could be the best...